pkg/snapshot/controllers/snapshotschedule_test.go

```diff
@@ -0,0 +1,133 @@
+package controllers
+
+import (
+	"testing"
+	"time"
+
+	snapv1 "github.com/kubernetes-incubator/external-storage/snapshot/pkg/apis/crd/v1"
+	stork_api "github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	fakeclient "github.com/libopenstorage/stork/pkg/client/clientset/versioned/fake"
+	"github.com/portworx/sched-ops/k8s/core"
+	k8sextops "github.com/portworx/sched-ops/k8s/externalstorage"
+	storkops "github.com/portworx/sched-ops/k8s/stork"
+	"github.com/stretchr/testify/assert"
+	gomock "go.uber.org/mock/gomock"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/resource"
+	kubernetes "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/rest/fake"
+)
+
+// TestCleanupErroredSnapshots tests the `cleanupErroredSnapshots` function that
+// cleans up snapshots in error state and older than a certain cutoff time period.
+func TestCleanupErroredSnapshots(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockSnap := NewMockOps(ctrl)
+	fakeStorkClient := fakeclient.NewSimpleClientset()
+	var unstructuredSerializer = resource.UnstructuredPlusDefaultContentConfig().NegotiatedSerializer
+	fakeRestClient := &fake.RESTClient{
+		NegotiatedSerializer: unstructuredSerializer,
+	}
+	fakeKubeClient := kubernetes.NewSimpleClientset()
+	core.SetInstance(core.New(fakeKubeClient))
+	storkops.SetInstance(storkops.New(fakeKubeClient, fakeStorkClient, fakeRestClient))
+
+	// Creating a dummy snapshotschedule object.
+	creationTime1 := metav1.NewTime(time.Now().Add(-30 * time.Minute))
+	creationTime2 := metav1.NewTime(time.Now().Add(-25 * time.Minute))
+	creationTime3 := metav1.NewTime(time.Now().Add(-10 * time.Minute))
+	creationTime4 := metav1.NewTime(time.Now().Add(-5 * time.Minute))
```

Comment:
let's create one more errored snapshot whose creation timestamp is within the timeout value. And the expectation is it should not get deleted.

Reply:
Done